ecosystem/nft-metadata-crawler/nft-metadata-crawler-utils/src/gcs.rs

```diff
@@ -0,0 +1,93 @@
+// Copyright © Aptos Foundation
+
+use anyhow::anyhow;
+use google_cloud_auth::token_source::TokenSource;
+use image::ImageFormat;
+use reqwest::{
+    header::{self, HeaderMap},
+    Client,
+};
+use serde_json::Value;
+
+pub async fn write_json_to_gcs(
+    ts: &dyn TokenSource,
+    bucket: String,
+    id: String,
+    json: Value,
+) -> anyhow::Result<String> {
+    let client = Client::new();
+    let filename = format!("json_{}.json", id);
+    let url = format!(
+        "https://storage.googleapis.com/upload/storage/v1/b/{}/o?uploadType=media&name={}",
+        bucket, filename
+    );
+    let json_string = json.to_string();
+
+    let res = client
+        .post(url)
+        .bearer_auth(ts.token().await?.access_token)
```

Comment:
what'll happen if access token expires? do we refresh it?

Reply:
Using the package that automatically takes the credentials and generates a new time on use. Changed this function ^